src/vmprof_main.h

```diff
@@ -207,6 +243,17 @@ static void sigprof_handler(int sig_nr, siginfo_t* info, void *ucontext)
     __sync_lock_release(&spinlock);
 #endif
 
+#ifdef VMPROF_LINUX
+    // SIGNAL ABUSE AHEAD
+    // On linux, the prof timer will deliver the signal to the thread which triggered the timer,
+    // because these timers are based on process and system time, and as such, are thread-aware.
+    // For the real timer, the signal gets delivered to the main thread, seemingly always.
+    // Consequently if we want to sample all threads, we need to forward the signal with tgkill.
+    if ((signal_type == SIGALRM) && (pid == tid)) {
+        broadcast_signal_for_threads(pid);
```

Comment:
I'm fine with this hack, I need to test it a little though. One question I have:

1) signal A (to main thread tid == pid == 1) arrives and uses tgkill to send a new signal to thread 2-3... (assuming there are 3 threads (1-3))
2) signal B to thread 2... OK
3) signal C to thread 1 (issuing more signals to thread 2 and 3)
4) signal D  to thread 3 ...

Will thread 3 get another signal initiated by signal C? Is it swallowed?
If the signal dispatching takes to long I think that should be recoginized...